### PR TITLE
Reverse group/member hierarchy

### PR DESCRIPTION
**Pull request by samiulsami**

### CODE REVIEW

File: inbox/helpers.go [hunk 1]
```diff
@@ -85,44 +85,28 @@ func getObjectIdentifierFromObjectInterface(object Object) (*ObjectIdentifier, e
...
-	member, err := getObjectIdentifierFromObjectInterface(childObject)
-	if err != nil {
-		return nil, err
-	}
-	if member == nil {
-		if !object.HasParentObject() {
-			return nil, nil
+	for ; obj != nil && obj.HasParentObject(); obj = obj.ParentObject {
+		if obj.ParentObject == nil {
+			return nil, fmt.Errorf("expected parent existence")
```
Comment (Imtiaz246) [inline inbox/helpers.go]: We don't need that as we already checking in the for loop

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: inbox/helpers_test.go [hunk 2]
```diff
@@ -36,15 +41,27 @@ func TestGetGroupAndAssociatedMemberIdentifier(t *testing.T) {
...
-		if !result.Member.HasParentObject() {
-			t.Errorf("member %s does not have parent object", result.Member.ObjectName)
-			return
+	prettyJSON, err := json.MarshalIndent(results, "  ", "   ")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	fmt.Printf("%v\n", string(prettyJSON))
+
```
Comment (Imtiaz246) [inline inbox/helpers_test.go]: remove this part also

File: inbox/helpers_test.go [hunk 3]
```diff
@@ -25,7 +26,11 @@ func TestGetGroupAndAssociatedMemberIdentifier(t *testing.T) {
 		},
 	}
 
-	results, err := getGroupAndAssociatedMemberIdentifier(jamesIdentifier, nil)
+	expectedParent := make(map[string]string)
+	expectedParent["database"] = "namespace"
+	expectedParent["namespace"] = "cluster"
+
```
Comment (Imtiaz246) [inline inbox/helpers_test.go]: expecteds should be on full form of the output

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: inbox/helpers_test.go (hunk 3 above)
Comment (samiulsami) [reply inbox/helpers_test.go]: Updated

File: inbox/helpers_test.go [hunk 4]
```diff
@@ -25,26 +27,96 @@ func TestGetGroupAndAssociatedMemberIdentifier(t *testing.T) {
...
 	}
 
-	if len(results) != 2 {
-		t.Error(fmt.Errorf("length of result doesn't match: expected: %v, got: %v", 3, len(results)))
-		return
+	expected := []GroupAndAssociatedMemberIdentifier{
+		{
+			Group: ObjectIdentifier{
+				ObjectName:     "database",
+				ObjectUniqueID: "1",
```
Comment (Imtiaz246) [inline inbox/helpers_test.go:40]: use the const vars

File: inbox/helpers_test.go [hunk 5]
```diff
@@ -25,26 +27,96 @@ func TestGetGroupAndAssociatedMemberIdentifier(t *testing.T) {
...
-			return
+		return results[i].Group.ObjectName < results[j].Group.ObjectName
+	})
+
+	sort.Slice(expected, func(i, j int) bool {
+		if expected[i].Group.ObjectName == expected[j].Group.ObjectName {
+			return expected[i].Member.ObjectName < expected[j].Member.ObjectName
+		}
+		return expected[i].Group.ObjectName < expected[j].Group.ObjectName
+	})
```
Comment (Imtiaz246) [inline inbox/helpers_test.go]: we don't need sorting here

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: inbox/helpers_test.go [hunk 6]
```diff
@@ -25,26 +27,96 @@ func TestGetGroupAndAssociatedMemberIdentifier(t *testing.T) {
...
+		t.Error(err)
 	}
+}
 
-	for _, result := range results {
-		fmt.Printf("group: %s, member: %s\n", result.Group.ObjectName, result.Member.ObjectName)
-		if !result.Member.HasParentObject() {
-			t.Errorf("member %s does not have parent object", result.Member.ObjectName)
-			return
+func isEqual(results, expected []GroupAndAssociatedMemberIdentifier) error {
```
Comment (Imtiaz246) [inline inbox/helpers_test.go]: isEqual is intended to return true or false. you can name it compare and return error

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

**Review by Imtiaz246 — APPROVED** [review]

**Pull request merged**